x-pack/plugins/observability_solution/observability_ai_assistant/server/service/index.ts

```diff
@@ -122,7 +124,13 @@ export class ObservabilityAIAssistantService {
     });
   }
 
-  init = once(async () => {
+  init = async () => {
+    if (this.initSuccessful) {
```

Comment:
I would rather encapsulate retry logic within the init function. That guarantees that this function cannot be run in parallel which may introduce new bugs.

Reply:
The issue in SDH that lead to this PR wouldn't have been solved by `init` doing retries internally though.
Since it required human intervention and then for `init` to run again.

Ideally we would make `init` idempotent but I'm not sure how to do that in a parallel safe way (at least not 100%) and I wonder if it would be too slow ("create if not exists"-checks).
What do you think?

Reply:
no it runs on every request. indeed, `once` doesn't deal with multiple Kibana instances. To do this properly we'd need a task I think. I do like the fact that we don't have to worry about initialization in many places, which would also imply handing back the error to the user.

Fwiw, the alerting plugin has a bunch of utility functions that have retry logic - maybe we can use those? They're in the same place as `createConcreteWriteIndex`.

Reply:
After speaking on Zoom, we decided to try and implement some way to "reset" the `once` setup of init:
When the service starts up, we set `init` to a function wrapped in `once` but which internally catches the error and if there is an error we set `init` again to the same function wrapped in once. 

This will ensure `init` only runs _once_ (even with multiple requests incoming) but it able to run again on the next set of requests if it fails.

Reply:
Fixed in https://github.com/elastic/kibana/pull/179346/commits/dd01cee3a876464b10a7aa91895af0bdb15e5f8a